### DISCUSSION TITLE
Test structs leaking into main library

### DISCUSSION THREAD

**Discussion by tpoliaw** [SeaQL/sea-orm#1073] [Q&A] [answered]
It looks like several types intended for tests are leaking into the library when compiled for other projects. This leads to surprising and unhelpful error messages for users such as `CakeFillingPrice` appearing in a list of suggested types
```
    error[E0277]: the trait bound `sea_orm::RelationDef: DeriveIden` is not satisfied
   --> src/main.rs:32:10
    |
32  |         .join_as(Alias::new("src"), JoinType::InnerJoin, entities::transaction::Relation::SourceAccount.def())
    |          ^^^^^^^ the trait `DeriveIden` is not implemented for `sea_orm::RelationDef`
    |
    = help: the following other types implement trait `DeriveIden`:
              Alias
              CakeFillingPrice
              Font
              Glyph
              Identity
              NullAlias
              RustKeyword
              SelectA
            and 63 others
    = note: required because of the requirements on the impl of `IntoIden` for `sea_orm::RelationDef`
```

It's generally only a problem when you're writing code that doesn't compile but this is most common when starting out with a new library when the user is often relying heavily on compiler suggestions and autocompletion to explore the API.

I have tried to have a look at a fix but while adding a `#[cfg(test)]` attribute to the `pub mod cfg_tests` declaration works to prevent the types being in the built library, it seems to also be preventing the module being available from tests. I'm not sure why but I'm guessing this might be behind the reason why it is the way it is. I'm happy to keep looking at it if there is not a reason they need to be available at runtime but thought I'd check before I keep digging.

Is there a reason for them to be exposed?

**Comment by billy1624** [comment, marked as answer]
Hey @tpoliaw, I will open an issue for this and share my thoughts :)

**Reply by billy1624** [reply]
Please check the related issues & PR
- https://github.com/SeaQL/sea-orm/issues/1076
- https://github.com/SeaQL/sea-orm/pull/1077
